File: Практика_самостоятельная работа/Python/2.18.py
import math

# Функция для вычисления z
def calculate_z(x, y):
    numerator = x + 2 + y / x**2  # Числитель
    denominator = y + 1 / math.sqrt(x**2 + 10)  # Знаменатель
    return numerator / denominator

# Функция для вычисления q
def calculate_q(x, y):
    return 7.25 * math.sin(x) - abs(y)

# Ввод значений x и y
x = float(input("Введите x: "))
y = float(input("Введите y: "))

# Вычисление и вывод результатов
z = calculate_z(x, y)
q = calculate_q(x, y)
print("z =", z)
print("q =", q)

File: Практика_самостоятельная работа/C++/2.18.cpp
#include <iostream>
#include <cmath>

using namespace std;

double calculate_z(double x, double y) {
    double numerator = x + 2 + y / (x * x); // Числитель
    double denominator = y + 1 / sqrt(x * x + 10); // Знаменатель
    return numerator / denominator;
}

double calculate_q(double x, double y) {
    return 7.25 * sin(x) - abs(y);
}

int main() {
    double x, y, z, q;

    cout << "Введите x: ";
    cin >> x;
    cout << "Введите y: ";
    cin >> y;

    z = calculate_z(x, y);
    q = calculate_q(x, y);

    cout << "z = " << z << endl;
    cout << "q = " << q << endl;

    return 0;
}
